### PR TITLE
feat(privateRepoIntegration): private and all repo integration

### PR DESCRIPTION
**Pull request by simran142002**
### Functionality:
Now we can integrate even private repos from organization page

### Solution:
added methods for private and all repo links in httpsService

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
ng serve -o
firebase emulators:start

### CODE REVIEW

File: src/app/body/github-link/github-link.component.html [hunk 1]
```diff
@@ -36,6 +36,34 @@ <h3>Team Name: <span class="bold">{{ teamName }}</span></h3>
                 <form *ngIf="!showClose" (ngSubmit)="submit()" #form="ngForm">
                     <ng-container *ngIf="!dataFetched">
                     <div class="row px-2">
+                        <ng-container>
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.html:39]: no use of ng-container

File: src/app/body/github-link/github-link.component.html [hunk 2]
```diff
@@ -36,6 +36,34 @@ <h3>Team Name: <span class="bold">{{ teamName }}</span></h3>
...
+                            <div class="col-md-12 py-0 my-0">
+                                <div class="btn-group">
+                                    <button class="btn dropdown-toggle box" type="button" id="button" data-toggle="dropdown" aria-haspopup="true"
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.html:42]: style can be moved to css file

File: src/app/body/github-link/github-link.component.html [hunk 3]
```diff
@@ -36,6 +36,34 @@ <h3>Team Name: <span class="bold">{{ teamName }}</span></h3>
...
+                                      <button class="dropdown-item" type="button" (click)="setLinkType('All')">
+                                        All
+                                      </button>
+                                    </div>
+                                  </div>
+                            </div>
+                        </ng-container>
+                        <ng-container *ngIf="linkType=='Private' || linkType == 'All'">
+                            <div class="col-md-12 my-0 py-0">
+                                <input type="text" class="form-control" [(ngModel)]="bearerToken"
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.html:63]: move style to css file

File: src/app/body/github-link/github-link.component.ts [hunk 4]
```diff
@@ -37,16 +39,36 @@ export class GithubLinkComponent implements OnInit {
...
         });
-      } 
-      else if(this.searchType == 'username') {
+      } else if (this.searchType == 'organisation') {
+        if (this.linkType=='Private'){
+          this.httpService.getGithubPrivateRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.ts:56]: remove console.log and subscribe should follow the rule of next, error, complete

File: src/app/body/github-link/github-link.component.ts [hunk 5]
```diff
@@ -37,16 +39,36 @@ export class GithubLinkComponent implements OnInit {
...
+            this.objData = data;
+            this.dataFetched = true;
+          })
+        } else if(this.linkType=='All') {
+          this.httpService.getGithubAllRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.ts:65]: remove console.log and subscribe should follow the rule of next, error, complete

File: src/app/body/github-link/github-link.component.ts [hunk 6]
```diff
@@ -56,10 +78,49 @@ export class GithubLinkComponent implements OnInit {
           this.dataFetched = true;
         });
       }
+      else if (this.searchType == 'username') {
+        if (this.linkType=='Private'){
+          this.httpService.getGithubPrivateRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.ts:87]: remove console.log and subscribe should follow the rule of next, error, complete

File: src/app/body/github-link/github-link.component.ts [hunk 7]
```diff
@@ -56,10 +78,49 @@ export class GithubLinkComponent implements OnInit {
...
+            this.objData = data;
+            this.dataFetched = true;
+          })
+        } else if(this.linkType=='All') {
+          console.log("check2");
+          this.httpService.getGithubAllRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
```
Comment (viv9k) [inline src/app/body/github-link/github-link.component.ts:97]: remove console.log and subscribe should follow the rule of next, error, complete

**Review by viv9k — CHANGES_REQUESTED** [review]
Please improve code quality

**Review by Pendyala-thanvi — APPROVED** [review]

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**